packages/hardhat-core/src/builtin-tasks/flatten.ts

```diff
@@ -135,5 +258,27 @@ task(
     types.inputFile
   )
   .setAction(async ({ files }: { files: string[] | undefined }, { run }) => {
-    console.log(await run(TASK_FLATTEN_GET_FLATTENED_SOURCE, { files }));
+    const [flattenedFile, metadata] = await run(
+      TASK_FLATTEN_GET_FLATTENED_SOURCE_AND_METADATA,
+      { files }
+    );
+
+    console.log(flattenedFile);
+
+    // Warn the user when SPDX licenses or pragma abicoder directives are not specified
+    if (metadata?.filesWithoutLicenses.length > 0) {
```

Comment:
If I understand correctly, the `?` operator is here because `metadata` could be null. But we can just use an object where the files are empty arrays.

Reply:
The check was to avoid a null value because in this part of the code the value might be null:

<img width="408" alt="image" src="https://github.com/NomicFoundation/hardhat/assets/18092467/bbb859fe-bd07-4bc4-864d-1eb95d538bd3">

But in this part of the code the value should be always populated so I can remove the check

Reply:
The check is needed because in this part of the code a null value could be returned:

<img width="440" alt="image" src="https://github.com/NomicFoundation/hardhat/assets/18092467/3378b5c8-7c7e-43cb-bb84-abc761301b03">